queries/reflect.go

```diff
@@ -71,11 +71,11 @@ func (q *Query) BindG(ctx context.Context, obj interface{}) error {
 // If you neglect checking the rows.Err silent failures may occur in your
 // application.
 //
-// Valid types to bind to are: *Struct, []*Struct, and []Struct. Keep in mind
-// if you use []Struct that Bind will be doing copy-by-value as a method
-// of keeping heap memory usage low which means if your Struct contains
-// reference types/pointers you will see incorrect results, do not use
-// []Struct with a Struct with reference types.
+// Valid types to bind to are: *Struct, []*Struct, and []Struct. If your Struct
+// contains reference types / pointers, these must implement the Scan interface.
+// See the StructField struct in queries/reflect_test.go for an example. If any
+// of the pointers in the struct are uninitialized, they will be initialized
+// automatically.
```

Comment:
This has a strange repercussion for the decimal type that sqlboiler supports. Currently the struct looks like this: https://github.com/volatiletech/sqlboiler/blob/master/types/decimal.go#L27-L29

The `NullDecimal` looks the same but behaves differently. If the pointer comes back from the database as `nil` inside the struct it must remain nil and should not be initialized to anything because then you've lost your nullability.

I did mention that this would be okay so long as it did not break the decimal type. I hope there's no actual problems here.